utils/data.py

```
@@ -0,0 +1,50 @@
+
+
+def write_json(path: Path, data: Any) -> None:
+    try:
+        path.parent.mkdir(parents=True, exist_ok=True)
+        tmp = path.with_suffix(path.suffix + ".tmp")
+        with tmp.open("w", encoding="utf-8") as f:
+            json.dump(data, f, ensure_ascii=False, indent=2)
+        tmp.replace(path)
+
+    except OSError as e:
+        logger.error("Failed to read JSON file: %s", path, exc_info=e)
```

Ruff (0.14.11)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)